### DISCUSSION TITLE
get note duration

### DISCUSSION THREAD

**Discussion by felipetovarhenao** [mido/mido#432] [Q&A]
Hi everybody,

I'm new to mido. I'm trying to figure out the easiest way to get the duration (i.e. delta time) between `'note_on'` messages — that is, not the duration of the note itself, but the time in between note onsets. 

When iterate through a `MidiFile()` and print the messages while on `.play()`, I see mostly `time=0` for all notes, which doesn't give me anything to work with. Any suggestions?

Thanks in advance!

**Comment by rdoursenaud** [comment]
`.play()` returns the messages at the time they should be played.
What you want is iterate over the messages without playing them.
